test/functional/tag_controller_test.rb

```diff
@@ -148,7 +148,12 @@ def setup
     json = ActiveSupport::JSON.decode(@response.body)
 
     assert_not_nil json
-    # assert more things to check this
+    assert_equal tags(:spectrometer).nodes.first.body_preview,
+    json.first[:preview]
+    assert_equal tags(:spectrometer).nodes.first.main_image_url,
+    json.first[:image] 
```

Comment:
OK - so we need to know how this is actually structured, so we can properly test it. I guess it's not just an array, so `.first` doesn't return anything. 

To debug, i added:

```ruby
puts json.inspect
``` 

on line 151. It shows an empty array. So `[].first` = `nil`. So... I added `puts @response.body.inspect` to see what happened.

...

It also returned "[]"! 

OK... I'm going to check that we're actually getting any nodes back at all.